string type
int32 sideLength
string color
